sentry/src/main/java/io/sentry/SentryClient.java

```diff
@@ -81,6 +81,22 @@ private boolean shouldApplyScopeData(
 
     options.getLogger().log(SentryLevel.DEBUG, "Capturing event: %s", event.getEventId());
 
+    if (event != null) {
+      if (event.getThrowable() != null
+          && options.containsIgnoredExceptionForType(event.getThrowable())) {
+        options
+            .getLogger()
+            .log(
+                SentryLevel.DEBUG,
+                "Event was dropped as the exception %s is ignored",
+                event.getThrowable().getClass());
```

Comment:
Since `event.getThrowable()` needs to unwrap the `ExceptionMechanismException`, I'd rather cache it in a local var instead of calling the method 3 times.

Reply:
I just moved the code but I can change it.